### ISSUE TITLE
Support type definitions based also on chain name and not just chain spec

### ISSUE THREAD

**Issue by ntn-x2** [polkadot-js/apps#8455]
The `@polkadot/api` package allows chain developer to export type definitions with:

```js
export interface OverrideBundleType {
    chain?: Record<string, OverrideBundleDefinition>;
    spec?: Record<string, OverrideBundleDefinition>;
}
```

In the apps, tho, only [match based on spec name](https://github.com/polkadot-js/apps/blob/master/packages/apps-config/src/api/spec/index.ts) is supported.
Is it possible to have the apps also support the `chain` definitions, using the same override rules that the api package uses?

**Comment by jacogr** [comment]
We used to have it - there was exactly 1 chain in there and on inspection was actually duplicated with spec. So it was removed due to non-use.

**Comment by ntn-x2** [comment]
Our chain relies on that to provide accurate type decoration. Is it too much of an effort and/or maintenance burden to add it back? Could you provide some pointers as to what would have to change so that I could also work on a PR to add it back, perhaps?

**Comment by jacogr** [comment]
Sorry for losing this is the stream of issues - what is the exact problem that you are experiencing that is solved by chain types? The reason for asking is that type injection is generally for (assuming everybody is on metadata 14) -

- RPC calls
- runtime calls
- historic information

Hence the question.

To add it back -

- chain folder alongside the spec folder
- add it to the build flatenning (e.g. the imported packages are being exported to JSON and then imported there - no code execution from the,)
- add it to the root exports for the types and it should be picked up

For interest sake, this is the PR that removed it (although obviously the use with the build step has changed since then) - https://github.com/polkadot-js/apps/pull/7956

**Comment by ntn-x2** [comment]
Hi, thanks for the details! We have an unfortunate situation in which two different deployments are identified by the same chain spec name, but different chain name. Hence, in our [type definitions](https://github.com/KILTprotocol/sdk-js/blob/develop/packages/type-definitions/src/index.ts#L128) we make use of the chain name to discriminate between them. Now, for the time being, we decorate the apps with only the chain spec that is unique, since we can't decorate two chains that have the same chain spec but in reality different runtimes.

**Comment by jacogr** [comment]
If you are willing to re-add will accept a PR. (Otherwise it would need to wait a bit until I get to this)

**Comment by ntn-x2** [comment]
Wasn't that hard. I gave it a go in https://github.com/polkadot-js/apps/pull/8594.
I will open a separate PR once this is merged to migrate KILT definitions to use chain names. I anyway tested that locally, and all our chains show the right runtime calls now 😊
